onnx/src/test/scala/vilcacora/onnx/TranslatorSuite.scala

```diff
@@ -0,0 +1,129 @@
+    assertEquals(Translator.translateNode(node), Right(Operation.Add("X", "Y", "Z")))
+  }
+
+  test("translateNode should translate a Cast node") {
+    val node = NodeProto(
+      opType = "Cast",
+      input = Seq("in"),
+      output = Seq("out"),
+      attribute = Seq(
+        AttributeProto(name = "to", i = 6, `type` = AttributeProto.AttributeType.INT),
+      ), // 6 = INT64
+    )
+    assertEquals(Translator.translateNode(node), Right(Operation.Cast("in", "out", DataType.Int64)))
+  }
+
+  // --- REWRITTEN TEST FOR SVMCLASSIFIER ---
+  test(
+    "translateNode should correctly translate an SVMClassifier node with floating point tolerance",
+  ) {
+    val node = NodeProto(
+      opType = "SVMClassifier",
+      input = Seq("features"),
+      output = Seq("label", "scores"),
+      attribute = Seq(
+        AttributeProto(name = "classlabels_ints", ints = Seq(0L, 1L)),
+        AttributeProto(name = "coefficients", floats = Seq(1.5f, -1.5f)),
+        AttributeProto(name = "kernel_params", floats = Seq(0.5f, 0.1f)), // Note: 0.1f is inexact
+        AttributeProto(name = "kernel_type", s = ByteString.copyFromUtf8("RBF")),
+        AttributeProto(name = "post_transform", s = ByteString.copyFromUtf8("SOFTMAX")),
+        AttributeProto(name = "rho", floats = Seq(0.9f)),
+        AttributeProto(name = "support_vectors", floats = Seq(1.1f, 2.2f)),
+        AttributeProto(name = "vectors_per_class", ints = Seq(1L, 1L)),
+      ),
+    )
+
+    val result = Translator.translateNode(node)
+    assert(result.isRight, "SVMClassifier translation failed unexpectedly")
+
+    // Define a small tolerance for comparing floating point numbers
+    val epsilon = 1e-6
+
+    result.foreach {
+      case op: Operation.SVMClassifier =>
+        assertEquals(op.classLabels, List(0L, 1L))
+        assertEquals(op.kernelType, SVMKernel.Rbf)
+        assertEquals(op.postTransform, PostTransform.Softmax)
+
+        // Use assertEquals with a tolerance (epsilon) for doubles
+        assertEqualsDouble(op.kernelParams.head, 0.5, epsilon)
+        assertEqualsDouble(op.kernelParams(1), 0.1, epsilon)
+        assertEqualsDouble(op.rho.head, 0.9, epsilon)
+
+        // Compare arrays of doubles element-by-element with tolerance
+        op.coefficients.zip(Array(1.5, -1.5)).foreach { case (actual, expected) =>
+          assertEqualsDouble(actual, expected, epsilon)
+        }
+        op.supportVectors.zip(Array(1.1, 2.2)).foreach { case (actual, expected) =>
+          assertEqualsDouble(actual, expected, epsilon)
+        }
+      case other => fail(s"Expected SVMClassifier but got $other")
+    }
+  }
+
+  // --- End-to-end tests using model files from resources ---
+
+  // Helper now includes the leading slash, which is the correct way to specify an absolute path on the classpath
+  def loadModelFromPath(modelPath: String): ModelProto = {
+    val bytes = Files.readAllBytes(Paths.get(modelPath))
+    ModelProto.parseFrom(bytes)
+  }
```

Comment:
Instead of loading from a filepath, you should use `getResourceAsInputStream`.

https://scala-native.org/en/latest/lib/javalib.html#embedding-resources